editor/utils/mobile/index.js

```diff
@@ -20,15 +20,32 @@ export const disableIsSidebarOpenedOnMobile = ( payload ) => (
  * Middleware
  */
 
-export const mobileMiddleware = ( { getState } ) => next => action => {
-	if ( action.type === 'REDUX_REHYDRATE' ) {
-		return next( {
-			type: 'REDUX_REHYDRATE',
-			payload: disableIsSidebarOpenedOnMobile( action.payload ),
-		} );
-	}
-	if ( action.type === 'TOGGLE_SIDEBAR' && action.isMobile === undefined ) {
-		return next( toggleSidebar( isMobile( getState() ) ) );
+export const mobileMiddleware = ( { getState, dispatch } ) => next => action => {
+	switch ( action.type ) {
+		case 'REDUX_REHYDRATE':
+			return next( {
+				type: 'REDUX_REHYDRATE',
+				payload: disableIsSidebarOpenedOnMobile( action.payload ),
+			} );
+		case 'TOGGLE_SIDEBAR':
+			if ( action.isMobile === undefined ) {
+				return next( toggleSidebar( isMobile( getState() ) ) );
+			}
+			break;
+		case 'BROWSER_RESIZE':
+			const previousIsMobile = isMobile( getState() );
+			if ( ! previousIsMobile ) {
+				next( action );
+				const nextIsMobile = isMobile( getState() );
+				// execute mobile changes when resizing to mobile and at the start (if on mobile).
+				if ( nextIsMobile ) {
```

Comment:
This means when we'll be back on "desktop" we'll loose our preference. What if we avoid the middleware entirely and just add a specific selector `isFixedToolbar` that returns the preference value on desktop and always false on mobile?

Reply:
Hi @youknowriad, Yes I agree that approach is simpler and more reliable, I switched this PR to use that approach.